File: EvaluateReversePolishNotation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> s = new Stack<>();

        for(String i : tokens){
            if(i.equals("+")){
                s.push(s.pop() + s.pop());
            }
            else if(i.equals("-")){ //Note String is an object not a primitive type, hence we need to use .equals()
                int a = s.pop();
                int b = s.pop();
                s.push(b - a);
            }
            else if(i.equals("*")){
                s.push(s.pop() * s.pop());
            }
            else if(i.equals("/")){
                int a = s.pop();
                int b = s.pop();
                s.push(b / a);
            }
            else{
                s.push(Integer.parseInt(i));
            }
        }
        return s.pop();
    }
    //Time: O(n)
    //Space: O(n)
}

File: EvaluateReversePolishNotation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for num in tokens:
            if num == "+":
                stack.append(stack.pop() + stack.pop())
            elif num == "-":
                num1 = stack.pop()
                num2 = stack.pop()
                stack.append(num2 - num1)
            elif num == "*":
                stack.append(stack.pop() * stack.pop())
            elif num == "/":
                num1 = stack.pop()
                num2 = stack.pop()
                stack.append(int(num2 / num1))
            else:
                stack.append(int(num))        
        return stack.pop()

# Time: O(n)
# Space: O(n)
